modules/javafx.web/src/main/java/com/sun/javafx/webkit/prism/WCGraphicsPrismContext.java

```diff
@@ -502,6 +507,51 @@ protected boolean shouldCalculateIntersection() {
         return false;
     }
 
+    private void fillOrDrawShapeMasked(Graphics g, Shape shape, Paint paint, BasicStroke stroke, boolean isFill) {
+
+        FilterContext filterContext = getFilterContext(g);
+        Rectangle clipRectangle = getClipRectNoClone();
+
+        PrDrawable imagePrDrawable = (PrDrawable)Effect.getCompatibleImage(filterContext,
+                g.getRenderTarget().getContentWidth(), g.getRenderTarget().getContentHeight());
+        Graphics g1 = imagePrDrawable.createGraphics();
+        state.apply(g1);
+        g1.setPaint(paint);
+        if(stroke != null) {
+            g1.setStroke(stroke);
+        }
+        if (isFill) {
+            g1.fill(shape);
```

Comment:
This will call the slower `fill(Shape)` method in all cases rather than the specialized `fillRect` or `fillRoundRect` method. Given all the other things that are done to draw a shape with a clip mask, I suspect that this is fine. One thing to consider is to pass in an enum instead of a boolean. The enum could say whether to use the specialized calls or the generic `fill` call. It's probably not worth the effort to make this change.

Reply:
other than paths only RoundRectangle2D is passed to this function. Shape can be checked if it is an instance of RoundRectangle2D and faster draw api can be called. added the same along with enum private to this class.